azure_costs_exporter/scrape.py

```diff
@@ -53,17 +53,27 @@ def convert_json_df(data):
 
     return df
 
-def extract_metrics_from_df(df, counter):
+def extract_metrics_from_df(df, metric_name, enrollment, registry):
     """
     Fill a counter object with the data from the data frame
     """
     groups = df.groupby(base_columns).sum()
-    for name, value in groups.iterrows():
-        meta = dict(zip(base_columns, name))
-        counter.record(value.ExtendedCost, **meta)
 
+    class AzureCollector(object):
+        def collect(self):
+            c = CounterMetricFamily(metric_name,
```

Comment:
For in-house code we normally put the first multi-line parameter on a new line as well:

    class AzureCollector(object):
        def collect(self):
            c = CounterMetricFamily(
                metric_name,
                "Costs billed to Azure Enterprise Agreement {}".format(enrollment),
                labels=base_columns)

It reduces jaggedness of the code and makes better use of the existing whitespace if lines get longer.